lib/hcloud/future.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'active_support/core_ext/string/inflections'
+
+module Hcloud
+  class Future < Delegator
+    # rubocop: disable Lint/MissingSuper
+    def initialize(client, target_class, id)
+      @target_class = target_class
+      @id = id
+      @__client = client
+    end
+    # rubocop: enable Lint/MissingSuper
+
+    def __getobj__
+      # pluralize class name and convert it to symbol
+      @__getobj__ ||= @__client.public_send(
+        "#{@target_class.name.downcase.demodulize}s".to_sym
```

Comment:
I think `demodulize.underscore.pluralize` would be better here instead of appending just an s. 
| Klass | downcase+s | underscore+pluralize |
|---------|------------------|------------------------------|
| FloatingIP | floatingips | floating_ips |
| Activity | activitys | activities |

Or maybe even use `demodulize.tabelize`

Docs: https://api.rubyonrails.org/v2.3.8/classes/ActiveSupport/CoreExtensions/String/Inflections.html

Reply:
Note: `demodulize.tabelize` converts a string into a db table-name which is exactly how the resource functions are named in the client. Otherwise the `downcase` would just turn the `FloatingIP` or `SSHKeys` to `floatingip` and `sshkeys` but it should be `floating_ip` and `ssh_key`.

The `tabelize` also pluralizes the input.